File: src/main/java/tscp/Move.java
package tscp;

/**
 * 1	capture 2	castle 4	en passant capture 8	pushing a pawn 2 squares 16	pawn
 * move 32	promote
 */
public class Move {

    byte from;
    byte to;
    byte promote;
    byte bits;

    Move() {
    }

    Move(byte from, byte to, byte promote, byte bits) {
        this.from = from;
        this.to = to;
        this.promote = promote;
        this.bits = bits;
    }

}

File: src/main/java/board/Move.kt
package board

/**
 * 1	capture 2	castle 4	en passant capture 8	pushing a pawn 2 squares 16	pawn
 * move 32	promote
 */
class Move {
    var from: Byte = 0
    var to: Byte = 0
    var promote: Byte = 0
    var bits: Byte = 0

    internal constructor()

    internal constructor(from: Byte, to: Byte, promote: Byte, bits: Byte) {
        this.from = from
        this.to = to
        this.promote = promote
        this.bits = bits
    }
}
